test/ValuedCSP.lean

```diff
@@ -0,0 +1,172 @@
+import Mathlib.Combinatorics.Optimization.ValuedCSP
+import Mathlib.Data.Rat.Order
+import Mathlib.Tactic.Positivity
+import Mathlib.Data.Fin.VecNotation
+import Mathlib.Data.Fin.Tuple.Curry
+
+/-!
+# VCSP examples
+
+This file shows two simple examples of General-Valued Constraint Satisfaction Problems (see
+[ValuedCSP definition](Mathlib/Combinatorics/Optimization/ValuedCSP.lean)).
+The first example is an optimization problem. The second example is a decision problem.
+-/
+
+def valuedCspTermOfUnary {D C : Type} [OrderedAddCommMonoid C]
+    {Γ : ValuedCsp D C} {ι : Type*} {f : D → C}
```

Comment:
Probably, out of the scope of this PR: `ValuedCsp` should be called `ValuedCSP`.

Reply:
Thanks for the suggestion! And yes, it should be done in a separate PR.

Reply:
#9701